narwhals/_spark_like/namespace.py

```diff
@@ -80,12 +81,17 @@ def nth(self: Self, *column_indices: int) -> SparkLikeExpr:
         )
 
     def lit(self: Self, value: object, dtype: DType | None) -> SparkLikeExpr:
-        if dtype is not None:
-            msg = "todo"
-            raise NotImplementedError(msg)
-
         def _lit(df: SparkLikeLazyFrame) -> list[Column]:
-            return [df._F.lit(value)]
+            column = df._F.lit(value)
+            if dtype:
+                import pyspark.sql.types as pyspark_types
+
+                native_dtype = narwhals_to_native_dtype(
+                    dtype, self._version, pyspark_types
+                )
+                column = column.cast(native_dtype)
```

Comment:
This should not load pyspark types directly anymore. You can do:
```suggestion
            if dtype:
                spark_like_types = df._native_dtypes

                native_dtype = narwhals_to_native_dtype(
                    dtype, self._version, spark_like_types
                )
                column = column.cast(native_dtype)
```